File: 0222-count-complete-tree-nodes/0222-count-complete-tree-nodes.py
# Definition for a binary tree node.
# class TreeNode:
#   def __init__(self, val=0, left=None, right=None):
#     self.val = val
#     self.left = left
#     self.right = right
class Solution:
  def countNodes(self, root: Optional[TreeNode]) -> int:
    if not root: return 0
    left = right = root
    lHt = rHt = 1
    while left.left:
      left = left.left
      lHt += 1
    while right.right:
      right = right.right
      rHt += 1
    if lHt == rHt: return 2**lHt-1
    return 1+self.countNodes(root.left)+self.countNodes(root.right)

File: 0222-count-complete-tree-nodes/0222-count-complete-tree-nodes.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *   int val;
 *   TreeNode left;
 *   TreeNode right;
 *   TreeNode() {}
 *   TreeNode(int val) { this.val = val; }
 *   TreeNode(int val, TreeNode left, TreeNode right) {
 *     this.val = val;
 *     this.left = left;
 *     this.right = right;
 *   }
 * }
 */
class Solution {
  public int countNodes(TreeNode root) {
    if (root == null) return 0;
    TreeNode left = root;
    TreeNode right = root;
    int lHt = 1;
    int rHt = 1;
    while (left.left != null) {
      left = left.left;
      lHt++;
    }
    while (right.right != null) {
      right = right.right;
      rHt++;
    }
    if (lHt == rHt) return (1 << lHt) - 1;
    return 1 + countNodes(root.left) + countNodes(root.right);
  }
}